### DISCUSSION TITLE
Errors about theory resolutions for thermal expansion...

### DISCUSSION THREAD

**Discussion by Wang-Yihu** [idaholab/moose#29616] [Q&A Modules: Solid mechanics] [answered]
Let's consider a problem about thermal expansion. A 2D circle plate with its center fixed are heated to 500 K. Its zero expansion temperature is 300 K. Its  thermal expansion coefficient is 2E-5 K^-1. Its radius is 50e-3 m. These conditions above are shown in the following figure:
![image](https://github.com/user-attachments/assets/b7243a0f-144a-4464-8a84-2c1ab6ac24b9)
And the govern equations and boundary conditions are:
![image](https://github.com/user-attachments/assets/02aa5e88-8a99-4347-8eff-925c755c0d1a)
Evidently, the theory solution for this problem is 
![image](https://github.com/user-attachments/assets/58fb9139-25b6-446d-8b56-540cb141fe67)
So I want to use this solution to examine MOOSE. So I edit this input file:
```
[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Mesh]
  [accg]
    type = AdvancedConcentricCircleGenerator
    ring_radii = '50.0e-3'
    ring_intervals = '200'
    ring_block_ids = '15 20'
    ring_block_names = 'inner_tri inner'
    external_boundary_id = 100
    external_boundary_name = 'right'
    num_sectors = 200
  []
  [name_dian1]
    type = BoundingBoxNodeSetGenerator
    input = accg
    bottom_left = '${fparse 0.0-0.000000001} ${fparse 0.0-0.000000001} ${fparse 0.0-0.000000001}'
    top_right = '${fparse 0.0+0.000000001} ${fparse 0.0+0.000000001} ${fparse 0.0+0.000000001}'
    new_boundary = 'fix_point'
  []
[]

[Materials]
  [./elasticity_tensor_steel]
    youngs_modulus = 210e9
    poissons_ratio = 0.28
    type = ComputeIsotropicElasticityTensor
  [../]
  [./stress]
    type = ComputeFiniteStrainElasticStress
  [../]
  [expansion_steel_3]
    type = ComputeThermalExpansionEigenstrain
    stress_free_temperature = 300.0
    thermal_expansion_coeff = 2.0e-5
    temperature = temp
    eigenstrain_name = thermal
  []
[]

[Modules/TensorMechanics/Master]
  [all]
    add_variables = true
    eigenstrain_names = thermal
    generate_output = 'vonmises_stress stress_xx strain_xx'
    strain = FINITE
    temperature = temp
  []
[]

[AuxVariables]
  [temp]
  []
[]

[AuxKernels]
  [changshu_wendu]
    type = ConstantAux
    variable = temp
    value = 500.0
  []
[]

[Executioner]
  type = Steady
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
[]


[BCs]
  [guding_x]
    type = DirichletBC
    variable = disp_x
    boundary = 'fix_point'
    value = 0.0
  []
  [guding_y]
    type = DirichletBC
    variable = disp_y
    boundary = 'fix_point'
    value = 0.0
  []
[]

[Postprocessors]
  [max_disp_x]
    type = NodalMaxValue
    variable = disp_x
  []
  [max_disp_y]
    type = NodalMaxValue
    variable = disp_y
  []
  [max_stress_xx]
    type = ElementExtremeValue
    variable = stress_xx
  []
  [max_strain_xx]
    type = ElementExtremeValue
    variable = strain_xx
  []
[]

[Outputs]
  exodus = true
[]

```
In theory, the max_disp_x and max_disp_y will be αΔTR = (2e-5 × 200 × 50e-3) m = 2e-4 m
the max_stress_xx will be very close to 0 Pa
the max_strain_xx will be αΔT =  2e-5 × 200 = 4e-3
![image](https://github.com/user-attachments/assets/bfac4cc9-2317-4754-9818-58792920674d)
The results diverge with the theory too much except the stress (close to 0) ! Could you tell me the reason? It seems that some error imported in the simulation I do not realize....

**Comment by GiudGiud** [comment, marked as answer]
Hello

You have to set a poisson ratio of 0 to retrieve this analytical solution. Otherwise, you are seeing additional expansion from Poisson's effect.

Guillaume

**Reply by Wang-Yihu** [reply]
You're right sir! But I'm still puzzled... It seems that in analytical solution, the Possions ratio cannot affect the final result... Why the solution from MOOSE is depended on the Possions ratio?
![image](https://github.com/user-attachments/assets/e921f271-7d88-4346-aa35-003faa0cdaf0)

**Reply by GiudGiud** [reply]
> It seems that in analytical solution, the Possions ratio cannot affect the final result

in your derivation you ignored the poisson ratio. you only looked at diagonal components of the stress tensor.
